File: src/util_paths.ts
// TypeScript twin of src/util_paths.py — personal-asset path resolution.
//
// As of PR #748's tenant scaffold, these helpers delegate to `tenantPath`
// (src/tenant.ts) which handles the three-level resolution
// (per-tenant / per-device / fleet-shared) plus the single-tenant legacy
// fallback. Single-tenant installs (no SUTANDO_TENANT_ID, no
// state/tenant.json) see no behavior change — `tenantPath` falls through
// to the legacy `$SUTANDO_PRIVATE_DIR/machine-<host>/<file>` shape when
// `tenant-default/` hasn't been provisioned.
//
// Two helpers preserved as the public surface:
//
//   personalPath(filename)          — per-machine state
//                                     (stand-identity.json, pending-questions.md).
//                                     Maps to tenantPath(..., 'device').
//   sharedPersonalPath(filename)    — fleet-shared state (notes, build_log).
//                                     Maps to tenantPath(..., 'shared').
//
// Special case: `stand-avatar.png` also lives under `<workspace>/assets/`
// in the public repo. The avatar fallback is preserved here.

import { existsSync } from 'node:fs';
import { join } from 'node:path';
import { tenantPath } from './tenant.js';

/** Per-machine resolver. Delegates to `tenantPath(..., 'device')`. */
export function personalPath(filename: string, workspace?: string): string {
	const ws = workspace ?? process.cwd();
	// Avatar fallback — stand-avatar.png ships under assets/ in the public
	// repo. Preserve the special case so a fresh install with no private
	// dir still resolves the asset.
	//
	// TODO(classification): the avatar is conceptually fleet-shared (every
	// Mac uses the same image), so it would arguably belong under
	// `sharedPersonalPath`. Moving it would subtly change resolution order
	// (`tenantPath(..., 'shared')` would check `tenant-<id>/` before the
	// assets/ fallback) which could mask a future per-tenant-avatar feature.
	// Leaving here for now; revisit if/when avatars get per-tenant
	// customization.
	if (filename === 'stand-avatar.png') {
		const inAssets = join(ws, 'assets', filename);
		if (existsSync(inAssets)) return inAssets;
	}
	return tenantPath(filename, 'device', ws);
}

/** Shared-across-fleet resolver. Delegates to `tenantPath(..., 'shared')`. */
export function sharedPersonalPath(filename: string, workspace?: string): string {
	return tenantPath(filename, 'shared', workspace);
}

File: src/util_paths.py
"""Resolve personal-asset paths with private-dir-first lookup.

As of PR #748+#753 (TypeScript) / this PR (Python), these helpers delegate
to `tenant_path` (src/tenant.py) which handles the three-level resolution
(per-tenant / per-device / fleet-shared) plus the single-tenant legacy
fallback. Single-tenant installs (no SUTANDO_TENANT_ID, no state/tenant.json)
see no behavior change.

Two public helpers preserved:

    personal_path(filename)        — per-machine state (stand-identity.json,
                                     pending-questions.md). Maps to
                                     tenant_path(..., 'device').
    shared_personal_path(filename) — fleet-shared state (notes, build_log).
                                     Maps to tenant_path(..., 'shared').

Special case: stand-avatar.png also lives under <workspace>/assets/ in the
public repo. The avatar fallback is preserved here.
"""
from __future__ import annotations
from pathlib import Path
from typing import Optional

# Local import so the file is usable both as `from util_paths import ...`
# (the existing convention in src/ scripts) and as `from src.util_paths
# import ...` from the repo root.
try:
    from tenant import tenant_path
except ImportError:
    from .tenant import tenant_path  # type: ignore[no-redef]

REPO_DIR = Path(__file__).resolve().parent.parent


def personal_path(filename: str, workspace: Optional[Path] = None) -> Path:
    """Resolve a personal-asset path. Delegates to tenant_path(..., 'device').

    Avatar fallback: stand-avatar.png ships under assets/ in the public
    repo. Preserve the special case so a fresh install with no private
    dir still resolves the asset.
    """
    ws = workspace if workspace is not None else REPO_DIR

    if filename == "stand-avatar.png":
        in_assets = ws / "assets" / filename
        if in_assets.exists():
            return in_assets

    return tenant_path(filename, "device", workspace=ws)


def shared_personal_path(filename: str, workspace: Optional[Path] = None) -> Path:
    """Resolve a shared-private path. Delegates to tenant_path(..., 'shared')."""
    return tenant_path(filename, "shared", workspace=workspace)
